app/models/recipe.rb

```diff
@@ -1,3 +1,15 @@
 class Recipe < ApplicationRecord
    validates_presence_of :name, :complexity, :genre
+
+   has_many :recipe_ingredients
+   has_many :ingredients, through: :recipe_ingredients
+
+   def total_cost
+      ingredients.sum(:cost)
+   end
+
+   def add_ingredient_to_recipe(ingredient_id)
+      ingredient = Ingredient.find_by(id: ingredient_id)
+      self.ingredients << ingredient
+   end
```

Comment:
This implementation is great. You're finding the ingredient and then associating it with the recipe. You do not necessarily have to do this in a separate model method. All of this work can be done in the controller since all you're doing is a find. I give you an example in the recipes controller.